### ISSUE TITLE
Create bPfile slowly

### ISSUE THREAD

**Issue by ashraful1980** [iceland2k14/bsgs#43]
Hi,
I have try to create bPfile as above  command: python3 create_bPfile_mcpu2.py 22000000000 bPfile.bin 4; noted the 24 hours it was created 20GB. Could please advice any other option how to making bPfile.bin 704.00GB instantly...

**Comment by iceland2k14** [comment]
Please don't create bPfile from below <v6. I will move all those and update the new v6.
You will only need bloomfile. bpfile will be almost discontinued in the updated version.

**Comment by ashraful1980** [comment]
bsgs Update:
Once again i have start the new program 6.20 pm; and let you know after 1 hour. The command is: python3 create_bPfile_mcpu2.py 22000000000 bPfile.bin 4
Desktop Configuration:
AMD Ryzen 3rd Gen 24 Core speed 4.2 GHz; 256 GB RAM speed 1333 MHz; 2X 3080 GPU.
![image](https://user-images.githubusercontent.com/52574658/139582966-fb264640-8377-46c3-84a3-dd25a3f1086f.png)


Keyhunt Update:
[+] Bloom filter for 51539607552 elements : 176671.80 MB
[+] Bloom filter for 2576980378 elements : 8833.59 MB
[+] Allocating 39321.00 MB for 2576980378 bP Points
Above Bloom filter making 5 Hours. And after run the program speed was found as attached...
![138476302-8ce5df28-c431-4cea-ad27-63c1a387be75](https://user-images.githubusercontent.com/52574658/139583194-58c6b855-fe96-4930-9e30-f8351edd66d8.png)

**Comment by ashraful1980** [comment]
Same as previous new version or uploaded file....

**Comment by iceland2k14** [comment]
NO. Please don't use this command "python3 create_bPfile_mcpu2.py 22000000000 bPfile.bin 4"
Use this ->
`python bsgs_create_bpfile_bloomfile.py 22000000000 bpfile.bin bloomfile.bin 12`

**Comment by ashraful1980** [comment]
Error!!!

![image](https://user-images.githubusercontent.com/52574658/139592166-786ee074-9e6b-449b-b7a7-3dca4e349b22.png)

**Comment by iceland2k14** [comment]
Now. Removed some extra functions. I will add them in future versions. 
Can you try now.

**Comment by ashraful1980** [comment]
Thanks and starting bpfile.bin and bloom.bin as bellow:

┌──(pc㉿DESKTOP-LGTRO1E)-[~/bsgs/v6_dll_bsgs]
└─$ python3 bsgs_create_bpfile_bloomfile.py 44000000000 bpfile.bin bloomfile.bin 48

[+] Program Running please wait...
[*] Number of elements should be a multiple of 1000*num_cpu. Automatically corrected it to nearest value: 43999968000
[+] Number of items required for Final Script : [bp : 209762] [bloom : 43999968000]
[+] Output Size of the files : [bp : 6712384 Bytes] [bloom : 237230022309 Bytes]
[+] Creating bpfile in range 1 to 209762
[+] File : bpfile.bin created successfully in 0.35 sec

===========================================================================
[+] Starting bloom file creation ... with False Positive probability : 1e-09
[+] bloom bits  : 1897840178472    size [226240 MB]
[+] bloom hashes: 30
[+] Initializing the bloom filters to null
[+] Number of CPU thread: 48
[+] Thread  0: 0x1 -> 0x36a33690
[+] Thread  2: 0x6d466d21 -> 0xa3e9a3b0
[+] Thread  3: 0xa3e9a3b1 -> 0xda8cda40
[+] Thread  1: 0x36a33691 -> 0x6d466d20
[+] Thread  4: 0xda8cda41 -> 0x1113010d0
[+] Thread  5: 0x1113010d1 -> 0x147d34760
[+] Thread  6: 0x147d34761 -> 0x17e767df0
[+] Thread  7: 0x17e767df1 -> 0x1b519b480
[+] Thread  9: 0x1ebbceb11 -> 0x2226021a0
[+] Thread  8: 0x1b519b481 -> 0x1ebbceb10
[+] Thread  10: 0x2226021a1 -> 0x259035830
[+] Thread  11: 0x259035831 -> 0x28fa68ec0
[+] Thread  13: 0x2c649c551 -> 0x2fcecfbe0
[+] Thread  12: 0x28fa68ec1 -> 0x2c649c550
[+] Thread  14: 0x2fcecfbe1 -> 0x333903270
[+] Thread  15: 0x333903271 -> 0x36a336900
[+] Thread  17: 0x3a0d69f91 -> 0x3d779d620
[+] Thread  16: 0x36a336901 -> 0x3a0d69f90
[+] Thread  18: 0x3d779d621 -> 0x40e1d0cb0
[+] Thread  20: 0x444c04341 -> 0x47b6379d0
[+] Thread  21: 0x47b6379d1 -> 0x4b206b060
[+] Thread  19: 0x40e1d0cb1 -> 0x444c04340
[+] Thread  22: 0x4b206b061 -> 0x4e8a9e6f0
[+] Thread  23: 0x4e8a9e6f1 -> 0x51f4d1d80
[+] Thread  24: 0x51f4d1d81 -> 0x555f05410
[+] Thread  25: 0x555f05411 -> 0x58c938aa0
[+] Thread  26: 0x58c938aa1 -> 0x5c336c130
[+] Thread  27: 0x5c336c131 -> 0x5f9d9f7c0
[+] Thread  28: 0x5f9d9f7c1 -> 0x6307d2e50
[+] Thread  30: 0x6672064e1 -> 0x69dc39b70
[+] Thread  29: 0x6307d2e51 -> 0x6672064e0
[+] Thread  31: 0x69dc39b71 -> 0x6d466d200
[+] Thread  34: 0x741ad3f21 -> 0x7785075b0
[+] Thread  32: 0x6d466d201 -> 0x70b0a0890
[+] Thread  33: 0x70b0a0891 -> 0x741ad3f20
[+] Thread  35: 0x7785075b1 -> 0x7aef3ac40
[+] Thread  36: 0x7aef3ac41 -> 0x7e596e2d0
[+] Thread  39: 0x852dd4ff1 -> 0x889808680
[+] Thread  37: 0x7e596e2d1 -> 0x81c3a1960
[+] Thread  38: 0x81c3a1961 -> 0x852dd4ff0
[+] Thread  40: 0x889808681 -> 0x8c023bd10
[+] Thread  41: 0x8c023bd11 -> 0x8f6c6f3a0
[+] Thread  42: 0x8f6c6f3a1 -> 0x92d6a2a30
[+] Thread  43: 0x92d6a2a31 -> 0x9640d60c0
[+] Thread  44: 0x9640d60c1 -> 0x99ab09750
[+] Thread  45: 0x99ab09751 -> 0x9d153cde0
[+] Thread  46: 0x9d153cde1 -> 0xa07f70470
[+] Thread  47: 0xa07f70471 -> 0xa3e9a3b00
[+] Completed [29331000]

Don't know when it will complete.....

**Comment by iceland2k14** [comment]
For me 3000000000 items took 2 hours. With 44 Billion items for you i guess it will be 1 day and maybe 6 hours roughly. 
If you see in between that the Completed progress is slow. Then you may kill it. 
I will think about any possible way to increase speed, but it will take time, not immediate.

**Comment by ashraful1980** [comment]
Maximum Capacity of bsgs:
![image](https://user-images.githubusercontent.com/52574658/140019435-4d6c1fed-b0b6-484c-a69c-acbe4a69112c.png)

Maximum Capacity of keyhunt:
![138476302-8ce5df28-c431-4cea-ad27-63c1a387be75](https://user-images.githubusercontent.com/52574658/140019519-ff86cd6b-3311-45e4-88ce-f0f4256d7326.png)

Keyhunt is 1000 time faster from bsgs.

**Comment by iceland2k14** [comment]
Yes keyhunt is faster. And all main functions will move to keyhunt once the python test is working fine. 
And this search is still 1 cpu not added multicpu search. Just think about this as prototype for playing around different things quickly.
